lib/seattleflu/id3c/cli/command/etl/redcap_det_scan.py

```diff
@@ -318,18 +318,34 @@ def create_specimen(record: dict, patient_reference: dict) -> tuple:
     """ Returns a FHIR Specimen resource entry and reference """
     def specimen_barcode(record: Any) -> str:
         """
-        Given a REDCap *record*, returns the barcode or corrected barcode if it
-        exists.
+        Given a REDCap *record*, returns the scanned barcode or entered
+        corrected barcode if it exists.
+
+        Raises AssertionError when barcodes within record do not match.
         """
-        barcode = record['return_utm_barcode'] or record['pre_scan_barcode']
+        known_problem_record_ids = ['210']
+        barcode_fields = ['return_utm_barcode', 'pre_scan_barcode', 'utm_tube_barcode_2', 'reenter_barcode']
+
+        for field in barcode_fields:
+            record[field] = record[field].strip().lower()
+
+        record_id = record['record_id']
+        barcode = record['return_utm_barcode']
+        pre_scan_barcode = record['pre_scan_barcode']
+
+        if record_id not in known_problem_record_ids and barcode and pre_scan_barcode:
```

Comment:
Why do we want to raise an AssertionError if the barcodes don't match but also ingest records with known mismatched barcodes? Is this just because the known barcodes have already entered our system?

Reply:
The `return_utm_barcode` is what the lab see's so that should be the true barcode. Once we confirm that is true, add to the known list of the problem record ids.